### PR TITLE
Update Cart Order and Checkout Routes

### PR DESCRIPTION
**Pull request by mikejolley**
This is part of #1938 and is targetting https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1957 which includes a refactor of the Store Cart API controllers.

This PR:

- Adds an `order_id` to the Cart Schema. This will contain the draft order ID during checkout, or 0 if an order has not yet been created.
- Adds a `POST cart/create-order` endpoint. This accepts an object with `billing_address`, `shipping_address` and `order_note` from the checkout and persists it to a draft order. Everything else (items, shipping, calculated totals) are taken from the **cart**. Returns an order object.
- Adds a placeholder for the `/checkout` route. We need to hash out further what is needed here for #1938 so we can process payments via gateways like the core checkout.

**Follow ups:**

- [ ] Update readme documentation
- [ ] Work on the `/checkout` endpoint - determine what it needs to accept and what it needs to return, then add processing code to the endpoint.

### How to test the changes in this Pull Request:

1. Unit tests should pass
2. Have a cart and POST to `cart/create-order`. Check the response returns a valid order with all of your items.
3. Then do a GET to `/cart`. Ensure it contains the order ID just created.

### CODE REVIEW

File: src/RestApi/StoreApi/Routes/CartCreateOrder.php [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+		return 'checkout-draft';
+	}
+
+	/**
+	 * Create order line items.
+	 *
+	 * @todo Knowing if items changed between the order and cart can be complex. Line items are ok because there is a
+	 * hash, but no hash exists for other line item types. Having a normalized set of data between cart and order, or
+	 * additional hashes, would be useful in the future and to help refactor this code. In the meantime, we're relying
+	 * on custom hashes in $this->draft_order to track if things changed.
```
[the remark this answers is not in the document]
Comment (mikejolley) [reply src/RestApi/StoreApi/Routes/CartCreateOrder.php]: There is nothing to action right now, this was a brain dump for future.

File: tests/php/RestApi/StoreApi/Routes/CartCreateOrder.php [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+		$this->assertArrayHasKey( 'created_via', $data );
+		$this->assertArrayHasKey( 'prices_include_tax', $data );
+		$this->assertArrayHasKey( 'events', $data );
+		$this->assertArrayHasKey( 'customer', $data );
+		$this->assertArrayHasKey( 'billing_address', $data );
+		$this->assertArrayHasKey( 'shipping_address', $data );
+		$this->assertArrayHasKey( 'customer_note', $data );
+		$this->assertArrayHasKey( 'items', $data );
+		$this->assertArrayHasKey( 'totals', $data );
+
```
[the remark this answers is not in the document]
Comment (mikejolley) [reply tests/php/RestApi/StoreApi/Routes/CartCreateOrder.php]: Which ones?  Address?

**Comment by mikejolley** [comment]
@Aljullu there should be `expires` and `timestamp`. It was changed a while back but the installer won't run again without a version change. Might need to manually fix that on your install.

Schema is:

```
CREATE TABLE {$wpdb->prefix}wc_reserved_stock (
				`order_id` bigint(20) NOT NULL,
				`product_id` bigint(20) NOT NULL,
				`stock_quantity` double NOT NULL DEFAULT 0,
				`timestamp` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
				`expires` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
				PRIMARY KEY  (`order_id`, `product_id`)
			) $collate;
```

File: tests/php/RestApi/StoreApi/Routes/CartCreateOrder.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (Aljullu) [reply tests/php/RestApi/StoreApi/Routes/CartCreateOrder.php]: Ah, never mind. Didn't read it correctly. :man_facepalming:

**Pull request merged**